packages/ui/react-ui-menu/src/components/Toolbar/Toolbar.tsx

```diff
@@ -8,17 +8,13 @@ import { Toolbar as NaturalToolbar } from '@dxos/react-ui';
 
 import { ToolbarDropdownMenu } from './ToolbarDropdownMenu';
 import { ToolbarToggleGroup } from './ToolbarToggleGroup';
-import { isMenu, isSeparator, type ToolbarProps } from './defs';
-import { type MenuAction } from '../../defs';
+import { type ToolbarActionGroupProperties, type ToolbarProps } from './defs';
+import { type MenuAction, type MenuItem, isMenuGroup, isSeparator, type MenuItemGroup } from '../../defs';
 import { ActionLabel } from '../ActionLabel';
+import { useMenu, useMenuItems } from '../MenuContext';
 
-const ToolbarItem = ({
-  iconSize,
-  action,
-  onAction,
-}: Pick<ToolbarProps, 'iconSize' | 'onAction'> & {
-  action: MenuAction;
-}) => {
+const ToolbarItem = ({ action }: { action: MenuAction }) => {
+  const { onAction, iconSize } = useMenu();
```

Comment:
is the menu context intended to be optional or required for this to work? it seems to me like required, in which case I would follow the pattern we have elsewhere of having the default value of the context be `null` rather than noops and `raise`ing if the context is missing

Reply:
Sure, I was considering refactoring this to use Radix’s scoped context, which does a better job of enforcing the use of the context.